R/correctionsataglance-data.R

```diff
@@ -1,357 +1,539 @@
             no_overlap_lines <- lines[!is_overlap]
             new_line <- no_overlap_lines[which(no_overlap_lines != overlap_lines)]
             
             if(length(new_line) != 0){ # overlap occurs somewhere in all existing lanes 
-              dataIn_df$line_num[n] <- min(new_line)
+              dataIn_df[["line_num"]][n] <- min(new_line)
             } else {
-              dataIn_df$line_num[n] <- max(dataIn_df$line_num) + 1
+              dataIn_df[["line_num"]][n] <- max(dataIn_df[["line_num"]]) + 1
             }
-            
           }
-
         }
         
         line_num <- 0 # work around irrelevant warnings from devtools::check()
         new_line_df <- dataIn_df %>% filter(line_num != 1)
         if(nrow(new_line_df) != 0){
           #new lines = max line number (not including first, bc it's not new)
-          addLines <- list(rectToShift = new_line_df$position, 
-                           lineNum = new_line_df$line_num,
-                           numNewLines = max(new_line_df$line_num) - 1) 
+          addLines <- list(rectToShift = new_line_df[["position"]], 
+                           lineNum = new_line_df[["line_num"]],
+                           numNewLines = max(new_line_df[["line_num"]]) - 1) 
         }
-        
       }
-      
     } 
-
     return(addLines)
   })
   
   lineUnlist <- unlist(fixLines)
   linesToAdd <- grep('numNewLines', names(lineUnlist))
-  numToAdd <- unname(lineUnlist[linesToAdd])
+  totalNewLines <- unname(lineUnlist[linesToAdd])
   
-  return(list(numToAdd = sum(numToAdd), dataShiftInfo = fixLines))
+  return(list(totalNewLines = sum(totalNewLines), dataShiftInfo = fixLines))
 }
 
-addYData <- function(allData, height, overlapInfo, dateLim){
-  empty_i <- which(unlist(lapply(allData, is.null)))
-  emptyDataNames <- names(allData)[empty_i]
-  
-  startH <- 100
-  for(d in seq(length(allData))){
-    len_data <- allData[[d]][['dataNum']]
-    len_data <- ifelse(is.null(len_data), 2, len_data)
+#' Parse CORR Approvals
+#'
+#' @description Reads and formats the approvals for use on the CORR report
+#' @param timeSeries The time series to get approvals from
+#' @param timezone The timezone to parse data into
+#' @param dateSeq The sequence of dates from the report start date to the report end date
+parseCorrApprovals <- function(timeSeries, timezone, dateSeq){
+  approvals <- timeSeries[['approvals']]
+  returnData <- list()
+  
+  if(!isEmptyOrBlank(approvals)){
+    approvals[['startTime']] <- flexibleTimeParse(approvals[['startTime']], timezone)
+    approvals[['endTime']] <- flexibleTimeParse(approvals[['endTime']], timezone)
+    approvals[['description']] <- paste("Approval:", approvals[['description']])
+    colors <- c()
     
-    ytop <- vector(mode = "numeric", length = len_data)
-    ybottom <- ytop
+    labels <- format(dateSeq, "%m/%Y")
     
-    ytop[] <- startH
-    ybottom[] <- startH - height
+    for (i in 1:length(approvals[['level']])) {
```

Comment:
I think this should probably be changed to an sapply, though the number of approvals is so small that it probably wouldn't make much of a difference. I can definitely make the change along with any other suggested changes though.